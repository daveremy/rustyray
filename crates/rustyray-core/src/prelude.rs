//! Prelude module for common imports

pub use crate::{
    // Core types
    Actor, ActorRef, ActorSystem,
    ObjectRef, TaskBuilder, TaskSystem,
    Result, RustyRayError,
};