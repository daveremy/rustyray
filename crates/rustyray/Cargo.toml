[package]
name = "rustyray"
version.workspace = true
edition.workspace = true
authors.workspace = true
license.workspace = true
repository.workspace = true

[dependencies]
rustyray-core = { workspace = true }
rustyray-macros = { workspace = true }

# Re-export key dependencies
tokio = { workspace = true }
async-trait = { workspace = true }
serde = { workspace = true, optional = true }
linkme = { workspace = true }
bincode = { workspace = true }

[features]
default = ["serde"]
serde = ["dep:serde", "rustyray-core/serde"]

[dev-dependencies]
anyhow.workspace = true
criterion.workspace = true