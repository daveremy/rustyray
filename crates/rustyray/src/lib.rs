//! RustyRay - A Rust implementation of Ray Core
//!
//! This is the main entry point that re-exports everything from the core
//! and macro crates, providing a unified API surface.

// Re-export core functionality
pub use rustyray_core::*;

// Re-export macros
pub use rustyray_macros::*;

// Prelude for common imports
pub mod prelude {
    pub use crate::{
        actor,
        // Core types
        actor::{Actor, ActorRef, ActorSystem},
        actor_methods,
        error::{Result, RustyRayError},

        main,
        object_ref::ObjectRef,
        // Macros
        remote,
        task::{TaskBuilder, TaskSystem},
    };

    pub use async_trait::async_trait;
}
