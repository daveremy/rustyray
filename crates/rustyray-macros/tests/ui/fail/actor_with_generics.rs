use rustyray::prelude::*;

// Actor with generics - should fail
#[rustyray::actor]
struct GenericActor<T> {
    value: T,
}

fn main() {}