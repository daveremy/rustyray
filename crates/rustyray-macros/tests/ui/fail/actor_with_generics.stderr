error: #[actor] does not support generic parameters yet
 --> tests/ui/fail/actor_with_generics.rs:5:20
  |
5 | struct GenericActor<T> {
  |                    ^^^

warning: unused import: `rustyray::prelude::*`
 --> tests/ui/fail/actor_with_generics.rs:1:5
  |
1 | use rustyray::prelude::*;
  |     ^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
